ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -203,23 +218,91 @@ func (h *FaultHandler) CheckNetworkBlackHolePort() func(http.ResponseWriter, *ht
-			field.RequestType: requestType,
-			field.Request:     request.ToString(),
-			field.Response:    responseBody.ToString(),
-		})
+		ctx := context.Background()
+		ctxWithTimeout, cancel := context.WithTimeout(ctx, requestTimeoutDuration)
+		defer cancel()
+
+		var responseBody types.NetworkFaultInjectionResponse
+		var statusCode int
+		chainName := fmt.Sprintf("%s-%s-%s", *request.TrafficType, *request.Protocol, strconv.FormatUint(uint64(*request.Port), 10))
+		running, cmdOutput, cmdErr := h.checkNetworkBlackHolePort(ctxWithTimeout, request, chainName, taskMetadata.TaskNetworkConfig.NetworkNamespaces[0].Path)
+
+		// We've timed out trying to check if the black hole port fault injection is running
+		if err := ctx.Err(); err != nil {
+			logger.Error("Request timed out", logger.Fields{
+				field.RequestType: requestType,
+				field.Request:     request.ToString(),
+				field.Error:       err,
+			})
+			statusCode = http.StatusInternalServerError
+			responseBody = types.NewNetworkFaultInjectionErrorResponse(requestTimedOutError)
+		} else if cmdErr != nil {
+			logger.Error("Unknown error encountered for request", logger.Fields{
+				field.RequestType: requestType,
+				field.Request:     request.ToString(),
+				field.Error:       cmdErr,
+			})
+			statusCode = http.StatusInternalServerError
+			responseBody = types.NewNetworkFaultInjectionErrorResponse(cmdOutput)
+		} else {
+			statusCode = http.StatusOK
+			if running {
+				responseBody = types.NewNetworkFaultInjectionSuccessResponse("running")
+			} else {
+				responseBody = types.NewNetworkFaultInjectionSuccessResponse("not-running")
+			}
+			logger.Info("[INFO] Successfully checked status for fault", logger.Fields{
+				field.RequestType: requestType,
+				field.Request:     request.ToString(),
+				field.Response:    responseBody.ToString(),
+			})
+		}
 		utils.WriteJSONResponse(
 			w,
-			http.StatusOK,
+			statusCode,
 			responseBody,
 			requestType,
 		)
 	}
 }
 
+// checkNetworkBlackHolePort will check if there's a running black hole port within the task network namespace based on the chain name and the passed in required request fields.
+// It does so by calling iptables linux utility tool.
+func (h *FaultHandler) checkNetworkBlackHolePort(ctx context.Context, request types.NetworkBlackholePortRequest, chain, netNs string) (bool, string, error) {
+	cmdString := fmt.Sprintf(iptablesChainExistCmd, chain, *request.Protocol, strconv.FormatUint(uint64(*request.Port), 10))
+	cmdList := strings.Split(cmdString, " ")
+
+	// For host mode, the task network namespace is the host network namespace (i.e. we don't need to run nsenter)
+	if netNs != "host" {
+		cmdList = append(strings.Split(fmt.Sprintf(nsenterCommandString, netNs), " "), cmdList...)
+	}
+
+	cmdOutput, err := h.runExecCommand(ctx, cmdList)
+	if err != nil {
+		if exitErr, eok := err.(*exec.ExitError); eok && exitErr.ExitCode() == 1 {
+			logger.Info("[INFO] Black hole port fault is not running", logger.Fields{
```

Comment:
curious - why we want to have two cases here? looks both return same result back.

Reply:
>eok && exitErr.ExitCode() == 1

You're referring to this? If so, the idea was to check for sure that the exit code is `1`

Reply:
We are not doing any special thing for this exit code, right? so we can log it for debugging directly?

Reply:
Sure I guess. I'll add it to the log statement